plugins/org.eclipse.xtext/src/org/eclipse/xtext/formatting2/regionaccess/ITextRegionRewriter.java

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+ * Copyright (c) 2015 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.formatting2.regionaccess;
+
+/**
+ * 
+ * Use {@link ITextRegionAccess#getRewriter()} to obtain an instance implementing this interface.
+ * 
+ * @author Moritz Eysholdt - Initial contribution and API
+ */
+public interface ITextRegionRewriter {
+
+	public void renderToAppendable(Iterable<? extends ITextReplacement> replacements, Appendable result);
+
+	public void renderToAppendable(ITextSegment segment, Iterable<? extends ITextReplacement> replacements,
```

Comment:
Appendable.append declares IOException to be thrown. Since the rewriter did not create the appendable, it probably cannot handle its error conditions. I suspect this method should declare IOException, too.